### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by alszla**

### CODE REVIEW

File: src/main/java/core/utils/Navigator.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package core.utils;
+
+import org.openqa.selenium.WebElement;
+
+/**
+ * Navigator class.
```
Comment (JosueRodriguezGarcia) [inline src/main/java/core/utils/Navigator.java]: Please, improve the class description

File: src/main/java/core/utils/Navigator.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public final class Navigator {
+
+    /**
+     * Private constructor for utility class.
```
Comment (JosueRodriguezGarcia) [inline src/main/java/core/utils/Navigator.java]: please improve the class description

File: src/main/java/core/utils/ReportGenerator.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import net.masterthought.cucumber.Configuration;
+import net.masterthought.cucumber.ReportBuilder;
+import net.masterthought.cucumber.Reportable;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * ReportGenerator class.
```
Comment (JosueRodriguezGarcia) [inline src/main/java/core/utils/ReportGenerator.java]: please improve the class description

File: src/test/java/hooks/Hooks.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import core.selenium.WebDriverManager;
+import cucumber.api.java.After;
+import cucumber.api.Scenario;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+
+/**
+ * Hooks class.
```
Comment (JosueRodriguezGarcia) [inline src/test/java/hooks/Hooks.java]: Please, improve the class description

File: src/test/java/runner/RunCukesTest.java [hunk 5]
```diff
@@ -1,10 +1,30 @@
 package runner;
 
+import core.utils.ReportGenerator;
+import cucumber.api.CucumberOptions;
+import cucumber.api.testng.AbstractTestNGCucumberTests;
+import org.testng.annotations.AfterTest;
+
 /**
  * RunCukesTest.
```
Comment (JosueRodriguezGarcia) [inline src/test/java/runner/RunCukesTest.java]: Please improve the class description

File: src/main/java/core/utils/Navigator.java (hunk 2 above)
Comment (alszla) [reply src/main/java/core/utils/Navigator.java]: Not a class, this is method description.

It is this short mainly to point out that this was a requirement imposed by checkstyle tool.

File: gradle.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
-BROWSER=Chrome
+BROWSER=Firefox
```
[the remark this answers is not in the document]
Comment (alszla) [reply gradle.properties]: Done

File: src/main/java/core/utils/ReportGenerator.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    /**
+     * Sets up a Configuration object from the reporting tool.
+     */
+    public static void generateReports() {
+        final File reportOutputDirectory = new File("target");
+        final List<String> jsonFiles = new ArrayList<>();
+        jsonFiles.add("target/cucumber.json");
+
```
[the remark this answers is not in the document]
Comment (alszla) [reply src/main/java/core/utils/ReportGenerator.java]: Done

File: src/main/java/core/utils/ReportGenerator.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        final String projectName = "oshpark";
+
```
[the remark this answers is not in the document]
Comment (alszla) [reply src/main/java/core/utils/ReportGenerator.java]: Done

File: src/main/java/core/utils/Navigator.java (hunk 1 above)
Comment (alszla) [reply src/main/java/core/utils/Navigator.java]: Ok. Fixed.

File: src/test/java/hooks/Hooks.java (hunk 4 above)
Comment (alszla) [reply src/test/java/hooks/Hooks.java]: Ready.

File: src/test/java/runner/RunCukesTest.java (hunk 5 above)
Comment (alszla) [reply src/test/java/runner/RunCukesTest.java]: Done

File: src/main/java/core/utils/Navigator.java (hunk 2 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/core/utils/Navigator.java]: You're right. I'll be careful.

**Pull request merged**